python/ray/data/_internal/stats.py

```diff
@@ -624,14 +618,6 @@ def clear_iteration_metrics(self, dataset_tag: str):
             if dataset_tag in self._last_iteration_stats:
                 del self._last_iteration_stats[dataset_tag]
```

Comment:
We still wanna clear the iteration stats on the StatsManager so that the async update thread could exit. 

But we don't clear the iteration metrics with the StatsActor.

Reply:
i believe there is a usage of `StatsManager.clear_iteration_metrics` in `test_stats.py` which should be removed: https://github.com/scottjlee/ray/blob/d7f7e0f58248b9de145949883989ab597f97a2da/python/ray/data/tests/test_stats.py#L1695

Reply:
Oh we still need to call that I think? Without calling `StatsManager.clear_iteration_metrics`, the update thread will not exit since we will always have the `_last_iteration_metrics` here: https://github.com/ray-project/ray/blob/master/python/ray/data/_internal/stats.py#L540-L542

Reply:
ah sorry, i confused myself with the method of the same name in `StatsActor`, i think you are right Ricky.